Cubical/Data/Int/MoreInts/QuoInt/Properties.agda

```diff
@@ -233,3 +235,82 @@ sign-pos-suc-· m (signed s (suc n)) = refl
 
 ·-injʳ : ∀ m n o → m · pos (suc o) ≡ n · pos (suc o) → m ≡ n
 ·-injʳ m n o p = ·-injˡ o m n (·-comm (pos (suc o)) m ∙ p ∙ ·-comm n (pos (suc o)))
+
+opIsoIntℤ : Iso (Int → Int → Int) (ℤ → ℤ → ℤ)
+opIsoIntℤ = (opIso isoIntℤ)
+
+_+'_ : ℤ → ℤ → ℤ
+_+'_ = Iso.fun opIsoIntℤ _+Int_
+
+sucℤ→Int : ∀ (n : ℤ) → sucInt (ℤ→Int n) ≡ ℤ→Int (sucℤ n)
+sucℤ→Int (pos n) = refl
+sucℤ→Int (neg zero) = refl
+sucℤ→Int (neg (suc zero)) = refl
+sucℤ→Int (neg (suc (suc n))) = refl
+sucℤ→Int (posneg i) = refl
+
+predℤ→Int : ∀ (n : ℤ) → predInt (ℤ→Int n) ≡ ℤ→Int (predℤ n)
+predℤ→Int (pos zero) = refl
+predℤ→Int (pos (suc n)) = refl
+predℤ→Int (neg zero) = refl
+predℤ→Int (neg (suc n)) = refl
+predℤ→Int (posneg i) = refl
+
+ℤ→Int+Int≡+ : ∀ (n m : ℤ) → (ℤ→Int n) +Int (ℤ→Int m) ≡ ℤ→Int (n + m)
+ℤ→Int+Int≡+ (pos zero) m =
+  (ℤ→Int (pos zero)) +Int (ℤ→Int m)
+  ≡⟨ sym (pos0+ (ℤ→Int m)) ⟩
+  ℤ→Int m
+  ≡⟨ cong ℤ→Int (sym (+-zeroʳ spos m)) ⟩
+  ℤ→Int (m + pos zero)
+  ≡⟨ cong ℤ→Int (+-comm m (pos zero)) ⟩
+  ℤ→Int (pos zero + m) ∎
+ℤ→Int+Int≡+ (pos (suc n)) m =
+  (ℤ→Int (pos (suc n))) +Int (ℤ→Int m)
+  ≡⟨ sym (sucInt+ (Int.pos n) (ℤ→Int m)) ⟩
+  sucInt ((Int.pos n) +Int (ℤ→Int m))
+  ≡⟨ cong sucInt (ℤ→Int+Int≡+ (pos n) m) ⟩
+  sucInt (ℤ→Int ((pos n) + m))
+  ≡⟨ sucℤ→Int ((pos n) + m) ⟩
+  ℤ→Int (sucℤ ((pos n) + m))
+  ≡⟨ cong ℤ→Int (sucℤ-+ˡ (pos n) m) ⟩
+  ℤ→Int ((pos (suc n)) + m) ∎
+ℤ→Int+Int≡+ (neg (suc zero)) m =
+  negsuc zero +Int ℤ→Int m
+  ≡⟨ sym (predInt+ (Int.pos zero) (ℤ→Int m)) ⟩
+  predInt ((Int.pos zero) +Int (ℤ→Int m))
+  ≡⟨ cong predInt (ℤ→Int+Int≡+ (neg zero) m) ⟩
+  predInt (ℤ→Int ((neg zero) + m))
+  ≡⟨ predℤ→Int ((neg zero) + m) ⟩
+  ℤ→Int (predℤ ((neg zero) + m))
+  ≡⟨ cong ℤ→Int (predℤ-+ˡ (neg zero) m) ⟩
+  ℤ→Int ((neg (suc zero)) +  m) ∎
+ℤ→Int+Int≡+ (neg (suc (suc n))) m =
+  negsuc (suc n) +Int ℤ→Int m
+  ≡⟨ sym (predInt+ (negsuc n) (ℤ→Int m)) ⟩
+  predInt ((negsuc n) +Int ℤ→Int m)
+  ≡⟨ cong predInt (ℤ→Int+Int≡+ (neg (suc n)) m) ⟩
+  predInt (ℤ→Int ((neg (suc n)) + m))
+  ≡⟨ predℤ→Int ((neg (suc n)) + m)⟩
+  ℤ→Int (predℤ ((neg (suc n)) + m))
+  ≡⟨ cong ℤ→Int (predℤ-+ˡ (neg (suc n)) m) ⟩
+  ℤ→Int ((neg (suc (suc n))) + m) ∎
+  -- uh, this looks like an extremely generic thing that could be done when defining any function from ℤ, thus only requiring defining it on `pos n` and `neg (suc n)` as one would kinda expect for integers; why don't we do that everywhere?`
```

Comment:
I agree that the `elim` given in `Base` is quite tautological and this could probably be added alongside it (or even replace it if it's not used anywhere else).

Reply:
It wasn't used, so I replaced it, and also used the new version to define the only complicated part of the PR, I think this is more readable now.